anchor/client/src/lib.rs

```diff
@@ -476,6 +478,21 @@ impl Client {
             message_validator,
         );
 
+        let dynamic_target_peers = config.network.target_peers.is_none();
+        if dynamic_target_peers {
+            let state = database.state();
+            let mut unique_subnets = HashSet::new();
+            for cluster_id in state.get_own_clusters() {
+                if let Some(cluster) = state.clusters().get_by(cluster_id) {
+                    let subnet = SubnetId::from_committee(cluster.committee_id(), SUBNET_COUNT);
+                    unique_subnets.insert(subnet);
+                }
+            }
+            let dyn_peers = min(60 + unique_subnets.len() * 3, 150);
```

Comment:
we can simplify this by counting the unique subnets using `itertools` [`unique`](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.unique) (which we already depend transitively:
 ```suggestion
           let unique_subnets = state
                .get_own_clusters()
                .iter()
                .filter_map(|id| state.clusters().get_by(id))
                .map(|s| SubnetId::from_committee(s.committee_id(), SUBNET_COUNT))
                .unique()
                .count();

            let dyn_peers = min(60 + unique_subnets.len() * 3, 150);
```